cloudmailer/cloudmailer.py

```diff
@@ -177,12 +171,15 @@ def getRoleAssignments(self, vms=None, projects=None):
         list(map(all_assignments.extend,filter(None,result_list)))
         return all_assignments
 
-    def getRoleAssignment(self, project):
-        users = list(set( map ( lambda z:
-            z.user['id'], filter ( lambda y:
-            y.scope['project']['id'] == project, filter(lambda x :
-            'project' in  x.scope , self.all_assignments )))))
-        return users
```

Comment:
Refactored to get_emails_for_project() the name of this function was confusing it was also easy to mix it up with `getRoleAssignments()` (plurals vs. singulars )

Reply:
In the original version, the first filtering keeps only the assignments for which 'project' is in the scope.
In your version you are skipping this check by accessing directly `item.scope['project']`. This might lead to error at runtime for accessing an undefined field, unless 'project' is always present.

Reply:
I changed it back to how the logic was before.